### PR TITLE
Remove lots of dead code

### PR DESCRIPTION
**Pull request by davidiw**
* Cleanup a lot of old DPN types
* Remove currency code from transactions
* Remove currency code from gas estimations in the VM
* Remove an old BCS canonicalization test
* Allow CI to run on any branch except main within aptos-core

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -252,9 +252,8 @@ workflows:
   # Build the Docker images and run Forge tests
   build-test-deploy:
     when:
-      or:
-        - equal: [auto, << pipeline.git.branch >>]
-        - equal: [canary, << pipeline.git.branch >>]
+      not:
+        equal: [main, << pipeline.git.branch >>]
```
Comment (zekun000) [inline .circleci/config.yml:256]: why remove canary?

File: types/src/account_state.rs [hunk 2]
```diff
@@ -37,17 +32,17 @@ impl AccountState {
             .map(|opt_ar| opt_ar.map(|ar| ar.address()))
     }
 
-    pub fn get_diem_account_resource(&self) -> Result<Option<DiemAccountResource>> {
-        self.get_resource::<DiemAccountResource>()
+    pub fn get_diem_account_resource(&self) -> Result<Option<AptosAccountResource>> {
```
Comment (zekun000) [inline types/src/account_state.rs:35]: we probably can remove this function?

**Review by zekun000 — APPROVED** [review]
https://github.com/aptos-labs/aptos-core/issues/396

File: .circleci/config.yml (hunk 1 above)
Comment (davidiw) [reply .circleci/config.yml:256]: when not main, it runs on all other branches

File: types/src/account_state.rs (hunk 2 above)
Comment (davidiw) [reply types/src/account_state.rs:35]: next pr we can delete even more legacy stuff

**Comment by davidiw** [comment]
There's a transient issue in CI. Unblocking this.

**Pull request merged**